### PR TITLE
introduced videojs as node module

### PR DESCRIPTION
**Pull request by theodortruffer**

### CODE REVIEW

**Comment by theodortruffer** [comment]
@iljalukin I integrated videojs as a node_module (v 7.5.5). Unfortunately I don't have a streaming server, so I can't test all functionalities. Could you test this for me on occasion?

The updating will be a lot easier like this. For minor updates just execute 'npm update' and for major versions change the version in /package.json and then 'npm update'.

File: classes/Util/class.xvmpVideoPlayer.php [hunk 1]
```diff
@@ -62,28 +62,31 @@ public function __construct($video, $embed = false) {
...
 		$tpl->addCss(ilViMPPlugin::getInstance()->getDirectory() . '/templates/default/video.css');
 
-                $tpl->addJavaScript(ilViMPPlugin::getInstance()->getDirectory() . '/vendor/video-js-7.5.4/video.js');
-                $tpl->addJavaScript(ilViMPPlugin::getInstance()->getDirectory() . '/vendor/video-js-7.5.4/videojs-contrib-quality-levels.js');
-                $tpl->addJavaScript(ilViMPPlugin::getInstance()->getDirectory() . '/vendor/video-js-7.5.4/videojs-hls-quality-selector.js');
-                $tpl->addCss(ilViMPPlugin::getInstance()->getDirectory() . '/vendor/video-js-7.5.4/video-js.css');
+		$tpl->addJavaScript(ilViMPPlugin::getInstance()->getDirectory() . '/node_modules/video.js/dist/video.min.js');
+		$tpl->addCss(ilViMPPlugin::getInstance()->getDirectory() . '/node_modules/video.js/dist/video-js.min.css');
+		$tpl->addJavaScript(ilViMPPlugin::getInstance()->getDirectory()
+			. '/node_modules/videojs-contrib-quality-levels/videojs-contrib-quality-levels.min.js');
```
Comment (iljalukin) [inline classes/Util/class.xvmpVideoPlayer.php]: this should contain the dist directory: . '/node_modules/videojs-contrib-quality-levels/dist/videojs-contrib-quality-levels.min.js');

File: classes/Util/class.xvmpVideoPlayer.php [hunk 2]
```diff
@@ -62,28 +62,31 @@ public function __construct($video, $embed = false) {
...
+		$tpl->addJavaScript(ilViMPPlugin::getInstance()->getDirectory()
+			. '/node_modules/videojs-hls-quality-selector/videojs-hls-quality-selector.min.js');
```
Comment (iljalukin) [inline classes/Util/class.xvmpVideoPlayer.php]: this should also contain the dist directory: . '/node_modules/videojs-hls-quality-selector/dist/videojs-hls-quality-selector.min.js');

**Comment by theodortruffer** [comment]
@iljalukin You're right, of course. It's fixed now, anything else?

**Comment by iljalukin** [comment]
Thanks for that, 
I think it should be possible to cache API requests for ViMP config options, since they don't change frequently, but are used in each video request. What's your opinion?

**Comment by theodortruffer** [comment]
I introduced a caching for the vimp version, for configs and for chapters. This should do a lot performance-wise (I hope). 

The config caching time (ttl) can be set in the plugin configuration. The chapters ttl is the same as the video metadata ttl. The version is cached until the cache is flushed.

@iljalukin Would you test it again? I will then merge it and release v1.1.0

**Comment by iljalukin** [comment]
@theodortruffer thank you, I have two more patches, that could be applied before the release

-  [66c4132239a60010c75985f0d77a5bc76149253b](https://github.com/iljalukin/ViMP/commit/66c4132239a60010c75985f0d77a5bc76149253b)  videojs-hls-quality-selector should be replaced through [videojs-http-source-selector](https://github.com/jfujita/videojs-http-source-selector), which has more options and the quality levels are shown in the right order
- [220e22bf12dc33ec701b69c3ecb94fd578a41697](https://github.com/iljalukin/ViMP/commit/220e22bf12dc33ec701b69c3ecb94fd578a41697) patch for "show own videos"-button

**Comment by theodortruffer** [comment]
@iljalukin I cherry-picked your commits and released version 1.1.0. Thanks a lot!

**Pull request merged**